src/common/hooks/use-set-network-mode.ts

```diff
@@ -0,0 +1,9 @@
+import { NetworkModes } from '@common/types/network';
+import { useMemo } from 'react';
+import { setCurrentNetworkMode } from '@common/app-helpers';
+
+export const useSetNetworkMode = (networkMode: NetworkModes) => {
+  useMemo(() => {
+    setCurrentNetworkMode(networkMode);
+  }, []);
```

Comment:
Shouldn't pass in networkMode? 🤔

Reply:
ah, no, we don't need to in this case. Normally we would but in this case we want it to run only one time. It is apart of this trick to set a recoil value on initial load from a value fetched on the server https://github.com/blockstack/explorer/blob/feat/improved-transactions/src/common/app-helpers.ts#L13